lib/web/apiserver_test.go

```diff
@@ -2002,6 +2003,106 @@ func TestGetMFADevices(t *testing.T) {
 	require.Equal(t, "otp", devices[0].Name) // default device name
 }
 
+func TestCreateAuthenticateChallenge_WithAuth(t *testing.T) {
+	t.Parallel()
+	env := newWebPack(t, 1)
+	proxy := env.proxies[0]
+
+	// Creates a user with a TOTP device, with second factor preference to OTP only.
+	pack := proxy.authPack(t, "llama@example.com")
+
+	endpoint := pack.clt.Endpoint("webapi", "mfa", "authnchallenge")
+	re, err := pack.clt.PostJSON(context.Background(), endpoint, url.Values{})
+	require.NoError(t, err)
+
+	var chal auth.MFAAuthenticateChallenge
+	err = json.Unmarshal(re.Bytes(), &chal)
+	require.NoError(t, err)
+	require.True(t, chal.TOTPChallenge)
+	require.Empty(t, chal.U2FChallenges)
+	require.Empty(t, chal.WebauthnChallenge)
+}
+
+func TestCreateAuthenticateChallenge_WithToken(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+	env := newWebPack(t, 1)
+	proxy := env.proxies[0]
+
+	ap, err := types.NewAuthPreference(types.AuthPreferenceSpecV2{
+		Type:         constants.Local,
+		SecondFactor: constants.SecondFactorOTP,
+	})
+	require.NoError(t, err)
+	err = env.server.Auth().SetAuthPreference(ctx, ap)
+	require.NoError(t, err)
+
+	// Create a user with a TOTP device.
+	username := "llama"
+	proxy.createUser(ctx, t, username, "root", "password", "some-otp-secret")
```

Comment:
Can't you do the same as above? Isn't the setup identical? Ie:

```suggestion
	// Creates a user with a TOTP device, with second factor preference to OTP only.
	pack := proxy.authPack(t, "llama@example.com")
```